### PR TITLE
Rotatable Db Secret Manager

### LINKED ISSUE #127: db secret rotation
secret rotation should be implemented at:

- db superuser secret defined [here](https://github.com/umccr/orcabus/blob/19b9bf194dc97c7171ea46f2edafac3ea09bdbe8/lib/workload/stateful/database/component.ts#L101)
	which then could remove the cdk-nag supression defined in the test [here](https://github.com/umccr/orcabus/blob/19b9bf194dc97c7171ea46f2edafac3ea09bdbe8/test/stateful/stateful-deployment.test.ts#L29-L35)
- microservice db role created by `orcabus-create-pg-login-role` (the secret manager here is out of IaC stack) #126

### PR DESCRIPTION
**Pull request by williamputraintan**
- Make Master and µ-app secret manager to start rotating
- Secret Manager for µ-app is now in CDK instead of created from lambda

ToDo after merge:
- delete µ-app secret manager that is created outside cdk

closes #127

### CODE REVIEW

File: lib/workload/stateless/postgres_manager/README.md [hunk 1]
```diff
@@ -52,7 +52,16 @@ There are 4 lambdas in this stack:
...
+    manager. The secret manager name is saved to `orcabus/${microserviceName}/rds-login-credential`.
+    You could make use of the static function called `formatDbSecretManagerName` in the `PostgresManagerStack` to get
+    the secret name, and you could pass this secret name to your app as an environment
+    variable when needed.
+
+    e.g.
+
+    ```typescript
+    const secretName = PostgresManagerStack.formatDbSecretManagerName('metadata_manager')
+    ```
```
Comment (victorskl) [inline lib/workload/stateless/postgres_manager/README.md:64]: Big 👍 up!

**Review by victorskl — APPROVED** [review]
Great stuff!

File: lib/workload/stateful/database/component.ts [hunk 2]
```diff
@@ -158,6 +163,20 @@ export class Database extends Construct {
       }),
     });
 
+    new sm.SecretRotation(this, 'MasterDbSecretRotation', {
+      application: sm.SecretRotationApplication.POSTGRES_ROTATION_SINGLE_USER,
+      // From default: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.DatabaseSecret.html#excludecharacters
+      excludeCharacters: '" %+~`#$&()|[]{}:;' + `<>?!'/@"\\")*`,
+      secret: dbSecret,
+      target: this.cluster,
+      automaticallyAfter: Duration.days(1),
```
Comment (reisingerf) [inline lib/workload/stateful/database/component.ts:172]: Should this not use the `secretRotationSchedule` parameter of the config?

**Review by reisingerf — COMMENTED** [review]
Nice!
Minor comment

File: lib/workload/stateful/database/component.ts (hunk 2 above)
Comment (williamputraintan) [reply lib/workload/stateful/database/component.ts:172]: Ah, miss reverting this after testing! Thanks

**Review by reisingerf — APPROVED** [review]

**Pull request merged**